src/pandas_profiling/model/expectation_algorithms.py

```diff
@@ -0,0 +1,70 @@
+def generic_expectations(name, summary, batch, *args):
+    batch.expect_column_to_exist(name)
+
+    if summary["n_missing"] == 0:
+        batch.expect_column_values_to_not_be_null(name)
+
+    if summary["p_unique"] == 1.0:
+        batch.expect_column_values_to_be_unique(name)
+
+    return name, summary, batch
+
+
+def numeric_expectations(name, summary, batch, *args):

```

Comment:
I think it's worth adding a few more numeric expectations here, like `expect_column_values_to_be_between` or `expect_column_min_to_be_between` and `expect_column_max_to_be_between`. You could add in a tolerance (e.g. adding 10% to a column max and subtracting 10% from a column min), but I think having the expectation be very exact is also OK.